src/GameState/Board.tsx

```diff
@@ -56,45 +57,102 @@ export const Board = (props: any) => {
+                top: `${item.y * boxMarbleSize}%`,
+                width: boxMarbleSizePercent,
+                height: boxMarbleSizePercent,
+                alignItems: 'center',
+                justifyContent: 'center',
+                zIndex: zIndex,
+                elevation: zIndex,
+              }}
+              key={item.x + ',' + item.y}
+            >
+              <Box
+                bg={
+                  marbleClickedCoordinates?.x === item.x &&
+                  marbleClickedCoordinates?.y === item.y
+                    ? BOARD_GREEN
+                    : ''
+                }
+              >
+                {item.value != 0 && (
+                  <Marble
+                    value={item.value}
+                    size={marbleColorSize}
+                    rowIndex={item.y}
+                    cellIndex={item.x}
+                    setMarbleClickedCoordinates={setMarbleClickedCoordinates}
+                    setMarbleClicked={props.setMarbleClicked}
+                    clicked={
+                      marbleClickedCoordinates?.x === item.x &&
+                      marbleClickedCoordinates?.y === item.y
+                    }
+                  />
+                )}
+              </Box>
+            </TranslateView>
+          );
+        })}
+        {console.log(marbleClickedCoordinates)}
+        {!props.preview && marbleClickedCoordinates && (
+          <>
+            {marbleClickedCoordinates.y > 0 && (
+              <DirectionControl
+                direction="N"
+                checkAndMoveMarble={props.checkAndMoveMarble}
+                coordinates={marbleClickedCoordinates}
+                boxMarbleSize={boxMarbleSize}
+                setMarbleClickedCoordinates={setMarbleClickedCoordinates}
+              />
+            )}
+
+            {marbleClickedCoordinates.x < 6 && (
+              <DirectionControl
+                direction="E"
+                checkAndMoveMarble={props.checkAndMoveMarble}
+                coordinates={marbleClickedCoordinates}
+                boxMarbleSize={boxMarbleSize}
+                setMarbleClickedCoordinates={setMarbleClickedCoordinates}
+              />
+            )}
+
+            {marbleClickedCoordinates.y < 6 && (
+              <DirectionControl
+                direction="S"
+                checkAndMoveMarble={props.checkAndMoveMarble}
+                coordinates={marbleClickedCoordinates}
+                boxMarbleSize={boxMarbleSize}
+                setMarbleClickedCoordinates={setMarbleClickedCoordinates}
+              />
+            )}
+
+            {marbleClickedCoordinates.x > 0 && (
+              <DirectionControl
+                direction="W"
+                checkAndMoveMarble={props.checkAndMoveMarble}
+                coordinates={marbleClickedCoordinates}
+                boxMarbleSize={boxMarbleSize}
+                setMarbleClickedCoordinates={setMarbleClickedCoordinates}
+              />
```

Comment:
The component is defined 4 times with only a tiny modification on direction! What about adding a function to decide which direction takes?